x-pack/filebeat/input/httpjson/request.go

```diff
@@ -88,12 +89,15 @@ type requestFactory struct {
 	password   string
 	log        *logp.Logger
 	encoder    encoderFunc
+	replace    string
 }
 
-func newRequestFactory(config *requestConfig, authConfig *authConfig, log *logp.Logger) *requestFactory {
+func newRequestFactory(config *requestConfig, configChain []chainsConfig, authConfig *authConfig, log *logp.Logger) []*requestFactory {
```

Comment:
Because the new argument is a slice, its name should be plural
```suggestion
func newRequestFactory(config *requestConfig, configChains []chainsConfig, authConfig *authConfig, log *logp.Logger) []*requestFactory {
```